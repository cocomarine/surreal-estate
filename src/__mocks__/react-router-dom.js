const reactRouterDom = jest.createMockFromModule("react-router-dom");

reactRouterDom.withRouter = ({ children }) => children;

module.exports = reactRouterDom;
