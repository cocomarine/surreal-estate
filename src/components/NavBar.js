import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import FacebookLogin from "react-facebook-login";
import "../styles/navbar.css";

const NavBar = ({ onLogin, userID, onLogout }) => {
  return (
    <div className="navbar">
      <img
        className="navbar-logo"
        src="/logo.png"
        alt="purple castle logo with 'surreal estate'"
      />
      <ul className="navbar-links">
        <li className="navbar-links-item">
          <Link to="/">View Properties</Link>
        </li>
        <li className="navbar-links-item">
          <Link to="add-property">Add a Property</Link>
        </li>
      </ul>
      <div className="login-button">
        {userID ? (
          <button className="logout-button" type="submit" onClick={onLogout}>
            SIGN OUT
          </button>
        ) : (
          <FacebookLogin appId="506761728339732" callback={onLogin} />
        )}
      </div>
    </div>
  );
};

NavBar.propTypes = {
  onLogin: PropTypes.func.isRequired,
  userID: PropTypes.string.isRequired,
  onLogout: PropTypes.func.isRequired,
};

export default NavBar;
