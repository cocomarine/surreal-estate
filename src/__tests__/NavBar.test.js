import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import NavBar from "../components/NavBar";

describe("NavBar", () => {
  const validProps = {
    onLogin: jest.fn(),
    userID: "test userID",
    onLogout: jest.fn(),
  };

  const setup = () => {
    render(
      <MemoryRouter>
        <NavBar {...validProps} />
      </MemoryRouter>
    );
  };

  it("renders correctly", () => {
    const rendered = renderer.create(
      <MemoryRouter>
        <NavBar {...validProps} />
      </MemoryRouter>
    );
    expect(rendered).toMatchSnapshot();
  });

  it("renders NavBar links correctly", () => {
    setup();

    expect(screen.getByText("View Properties")).toBeInTheDocument();
    expect(screen.getByText("Saved Properties")).toBeInTheDocument();
    expect(screen.getByText("Add a Property")).toBeInTheDocument();
  });
});
