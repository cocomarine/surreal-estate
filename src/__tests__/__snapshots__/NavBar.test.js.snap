// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<div
  className="navbar"
>
  <img
    alt="purple castle logo with 'surreal estate'"
    className="navbar-logo"
    src="/logo.png"
  />
  <ul
    className="navbar-links"
  >
    <li
      className="navbar-links-item"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        View Properties
      </a>
    </li>
    <li
      className="navbar-links-item"
    >
      <a
        href="/saved-properties"
        onClick={[Function]}
      >
        Saved Properties
      </a>
    </li>
    <li
      className="navbar-links-item"
    >
      <a
        href="/add-property"
        onClick={[Function]}
      >
        Add a Property
      </a>
    </li>
  </ul>
  <div
    className="login-button"
  >
    <button
      className="logout-button"
      onClick={[MockFunction]}
      type="submit"
    >
      SIGN OUT
    </button>
  </div>
</div>
`;
